lib/engine/game/g_1854/step/waterfall_auction.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/waterfall_auction'
+
+module Engine
+  module Game
+    module G1854
+      module Step
+        class WaterfallAuction < Engine::Step::WaterfallAuction
+          def actions(entity)
+            return [] if @game.need_auction_or
+
+            super
+          end
+
+          def setup
+            super
+            original_companies = @game.initial_auction_companies.sort_by(&:value)
+            @companies = @game.companies.select { |company| !company.owned_by_player? && !company.closed? }
+            @cheapest = original_companies.first
```

Comment:
Do you need to override this method? I don't believe that the initial auction is much different from an 1830-style auction.

Reply:
In the normal waterfall auction, when all players pass, but the auction isn't finished yet, it calls `payout_companies`, but in 1854, any of the local 1-6 minors need to actually operator, so we need to keep track of where the auction was before it was we run the OR.

Reply:
Ah, I think I see. You're starting a new auction round, so the modification to the `@companies` instance variable is not maintained.
It would be cleaner to override `@game.initial_auction_companies` to return unsold companies, instead of doing this here.
Or this might work – `@cheapest` should be set correctly in the superclass method, so doesn't need to be redefined here (I've removed the check for `company.closed?`, is that needed?):
```suggestion
            super
            @companies.reject! { |company| company.owned_by_player? }
```

Reply:
I moved this to the game logic as suggested, that does seem cleaner.  I did leave the `.closed?` check because the 1-6 minors are coded as companies that assign ownership of the corresponding minor, then close immediately.  This is to prevent them from appearing back in the pool again, as they are not owned by a player at that time.